### ISSUE TITLE
Still cpplint failures ? 

### ISSUE THREAD

**Issue by mhdawson** [nodejs/abi-stable-node#159]
Still seem to be getting a number of ccplint messages, From my attempt at adding the linting to the job:

https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-commit-linux-abi-mdawson/6/nodes=ubuntu1404-64/console

```
src/node_jsvmapi.cc:1:  Complex multi-line /*...*/-style comment found. Lint may give bogus warnings.  Consider replacing these with //-style comments, with #if 0...#endif, or with more clearly structured multi-line comments.  [readability/multiline_comment] [5]
src/node_jsvmapi.cc:2:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.cc:2:  Use operator && instead of and  [readability/alt_tokens] [2]
src/node_jsvmapi.cc:3:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.cc:3:  Use operator && instead of and  [readability/alt_tokens] [2]
src/node_jsvmapi.cc:4:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.cc:5:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.cc:5:  Use operator && instead of and  [readability/alt_tokens] [2]
src/node_jsvmapi.cc:5:  Use operator ! instead of not  [readability/alt_tokens] [2]
src/node_jsvmapi.cc:6:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.cc:7:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.cc:7:  Use operator && instead of and  [readability/alt_tokens] [2]
src/node_jsvmapi.cc:8:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.cc:9:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.cc:770:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:771:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:785:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
src/node_jsvmapi.cc:1072:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1073:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1080:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1101:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1102:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1547:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1548:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1562:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1563:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1577:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1578:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1592:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1593:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1733:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:1858:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.cc:2034:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.h:1:  Complex multi-line /*...*/-style comment found. Lint may give bogus warnings.  Consider replacing these with //-style comments, with #if 0...#endif, or with more clearly structured multi-line comments.  [readability/multiline_comment] [5]
src/node_jsvmapi.h:2:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.h:2:  Use operator && instead of and  [readability/alt_tokens] [2]
src/node_jsvmapi.h:3:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.h:3:  Use operator && instead of and  [readability/alt_tokens] [2]
src/node_jsvmapi.h:4:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.h:5:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.h:5:  Use operator && instead of and  [readability/alt_tokens] [2]
src/node_jsvmapi.h:5:  Use operator ! instead of not  [readability/alt_tokens] [2]
src/node_jsvmapi.h:6:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.h:7:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.h:7:  Use operator && instead of and  [readability/alt_tokens] [2]
src/node_jsvmapi.h:8:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.h:9:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/node_jsvmapi.h:236:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.h:237:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.h:380:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.h:393:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.h:394:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.h:416:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi.h:417:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/node_jsvmapi_types.h:0:  No #ifndef header guard found, suggested CPP variable is: SRC_NODE_JSVMAPI_TYPES_H_  [build/header_guard] [5]
test/addons-abi/test_buffer/test_buffer.cc:126:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
test/addons-abi/test_object/test_object.cc:91:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
Total errors found: 57
```

**Comment by mhdawson** [comment]
@digitalinfinity

**Comment by digitalinfinity** [comment]
Looking

**Comment by digitalinfinity** [comment]
Trying to repro locally but from a quick glance at the logs, I suspect a later change might have added the BOM to node_jsvmapi.h and node_jsvmapi.cc - I'll send out a PR to remove but I think that we might want to look at changing the linter to be tolerant of the BOM

**Comment by digitalinfinity** [comment]
Submitted #160 to fix these errors- can you try your CI against that branch and see if it looks green? We can use both the PR as a test of the CI and the CI as a test of the PR 😄

**Comment by digitalinfinity** [comment]
Landed as 93fe9fb
